.gitlab/source_test/golang_deps_diff.yml

```diff
@@ -25,6 +25,21 @@ golang_deps_commenter:
   rules: # this should only run on dev branches
     - !reference [ .except_main_or_release_branch ]
   needs: ["golang_deps_diff"]
-  script:
+  script: # ignore error message about no PR, because it happens for dev branches without PRs
     - echo "${CI_COMMIT_REF_NAME}"
-    - pr-commenter --for-pr="${CI_COMMIT_REF_NAME}" --header="Go Package Import Differences" --infile deps-report.md
+    - |
+      set +e
+      out=$(pr-commenter --for-pr="${CI_COMMIT_REF_NAME}" --header="Go Package Import Differences" --infile deps-report.md 2>&1)
+      exitcode=$?
+      set -e
+      if [ -n "${out}" ]; then
+        if [ $exitcode -eq 0 ]; then
+          echo $out
+        else
+          echo $out >&2
+        fi
+      fi
+      if [ "${out}" != "${out/invalid request: no pr found for this commit}" ]; then
```

Comment:
```suggestion
      if [ "${out}" != "${out:-invalid request: no pr found for this commit}" ]; then
```

Reply:
I don't think `:-` operator does what we want, it seems to provide a default value if unset? I want to remove `invalid request: no pr found for this commit` from the string.

Reply:
So if the results are not equal, then it means the error string was present.